### PR TITLE
cleanup, baseValidator and new moduleStyle

### PR DESCRIPTION
**Pull request by wpdebruin**
as discussed today on Slack
I changed the ValidationData type in ValidationError from string to Struct since we have several validators which are defined as struct.
This should fix the problem with customMessages, because now ALL messageReplacements for {ValidationData} will be skipped if this is a struct.

### CODE REVIEW

File: models/validators/BaseValidator.cfc [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+	property name="name";
+
+    /**
+     * check if field is NOT null and has length
+     * 
+     * @targetValue the value to check for nullness/emptyness
+     * @return boolean
+     */
+    boolean function isNullOrEmpty(required any targetValue){
```
Comment (lmajano) [inline models/validators/BaseValidator.cfc:17]: Hot dang!! This is an awesome method!

File: models/validators/BaseValidator.cfc [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright since 2020 by Ortus Solutions, Corp
+ * www.ortussolutions.com
+ * ---
+ * The field under validation must be yes, on, 1, or true. This is useful for validating "Terms of Service" acceptance.
+ */
+component {
```
Comment (lmajano) [inline models/validators/BaseValidator.cfc:7]: You must add `accessors=true` if not ACF won't grab it.

**Review by lmajano — CHANGES_REQUESTED** [review]
Few changes and we are ready to merge

File: ModuleConfig.cfc [hunk 3]
```diff
@@ -12,17 +12,16 @@ component {
 	this.webURL                     = "http://www.ortussolutions.com";
```
Comment (lmajano) [inline ModuleConfig.cfc:12]: Make sure you add our `cfformat` rules and run them.

File: models/validators/BaseValidator.cfc (hunk 2 above)
Comment (wpdebruin) [reply models/validators/BaseValidator.cfc:7]: done

Comment (wpdebruin) [reply models/validators/BaseValidator.cfc:7]: And I found some accessors typo's in several files

File: ModuleConfig.cfc (hunk 3 above)
Comment (wpdebruin) [reply ModuleConfig.cfc:12]: done

**Comment by lmajano** [comment]
great job @wpdebruin !

**Pull request merged**